pkg/oc/cli/cmd/create/deploymentconfig.go

```diff
@@ -32,131 +30,87 @@ var (
 )
 
 type CreateDeploymentConfigOptions struct {
-	DC     *appsapi.DeploymentConfig
-	Client appsinternalversion.DeploymentConfigsGetter
+	*CreateSubcommandOptions
 
-	DryRun bool
+	Image string
+	Args  []string
 
-	OutputFormat string
-	Out          io.Writer
-	Printer      ObjectPrinter
+	Client appsv1client.DeploymentConfigsGetter
 }
 
 // NewCmdCreateDeploymentConfig is a macro command to create a new deployment config.
-func NewCmdCreateDeploymentConfig(name, fullName string, f kcmdutil.Factory, out io.Writer) *cobra.Command {
-	o := &CreateDeploymentConfigOptions{Out: out}
-
+func NewCmdCreateDeploymentConfig(name, fullName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
```

Comment:
does it need a factory or just the restclientgetter?  narrow scope where easily possible.